static/app/components/stateLayer.tsx

```diff
@@ -0,0 +1,67 @@
+import isPropValid from '@emotion/is-prop-valid';
+import styled from '@emotion/styled';
+
+import {defined} from 'sentry/utils';
+
+interface StateLayerProps extends React.HTMLAttributes<HTMLSpanElement> {
+  color?: string;
+  higherOpacity?: boolean;
```

Comment:
Not a huge fan of needing to pass this, but I see why it's necessary and can't think of a way around it. Seems like an easy way for things to become inconsistent 😕

Reply:
Yeah I agree, lots of room for inconsistencies. One thing that I considered but decided against is to automate this by using the `color` package to check the contrast level between the current text color and the background color. If the contrast is below a certain amount then we enforce higher opacity. But that would involve a bunch of style queries and seems too bloated for its purpose.